types/twitch-js/index.d.ts

```diff
@@ -0,0 +1,175 @@
+    channels?: string[];
+    connection?: {
+        server?: string,
+        port?: number,
+        reconnect?: boolean,
+        maxReconnectAttempts?: number,
+        maxReconnectInterval?: number,
+        reconnectDecay?: number,
+        reconnectInterval?: number,
+        secure?: boolean,
+        timeout?: number,
+    };
+    identity?: {
+        username?: string,
+        password?: string,
+    };
+    options?: {
+        clientId?: string,
+        debug?: boolean,
+        commandTimeout?: number,
+    };
+    logger?: Logger;
+}
+
+export interface Message {
+    raw: string;
+    tags: TagsCollection;
+    prefix: string|null;
+    command: string;
+    params: string[];
+}
+
+export interface Emote {
+    code: string;
+    id: number;
+}
+
+export interface TagsCollection {
+    [key: string]: string|boolean|number|null;
+}
+
+export interface Utilities {
+    levenshtein: (s1: string, s2: string, caseSensitive: boolean) => number;
+    raffle: Raffle;
+    symbols: (line: string) => number;
+    uppercase: (line: string) => number;
+}
+
+export interface Raffle {
+    init: (channel: string) => void;
+    enter: (channel: string, username: string) => void;
+    leave: (channel: string, username: string) => boolean;
+    pick: (channel: string) => string|null;
+    reset: (channel: string) => void;
+    count: (channel: string) => number;
+    isParticipating: (channel: string, username: string) => boolean;
+}
+
+export class Client extends EventEmitter {
+    channels: string[];
+    clientId: string|null;
+    currentLatency: number;
+    emotes: string;
+    emotesets: {[key: string]: Emote[]};
+    globaluserstate: TagsCollection;
+    lastJoined: string;
+    latency: Date;
+    log: Logger;
+    maxReconnectAttempts: number;
+    maxReconnectInterval: number;
+    moderators: {[key: string]: string[]};
+    opts: ClientOptions;
+    pingLoop: number|null;
+    pingTimeout: number|null;
+    port: string;
+    reason: string;
+    reconnect: boolean;
+    reconnectDecay: number;
+    reconnectInterval: number;
+    reconnectTimer: number;
+    secure: boolean;
+    server: string;
+    username: string;
+    userstate: TagsCollection;
+    wasCloseCalled: boolean;
+    ws: WebSocket|null;
+
+    readonly utils: Utilities;
+
+    constructor(opts?: ClientOptions);
+
+    /// Base Methods
+    api(options?: ApiOptions, cb?: RequestCallback): Promise<any>;
+    getChannels(): string[];
+    getOptions(): ClientOptions;
+    getUsername(): string;
+    handleMessage(message: Message): void;
+    isMod(channel: string, username: string): boolean;
+    readyState(): 'CONNECTING'|'OPEN'|'CLOSING'|'CLOSED';
+
+    /// Commands
+    action(channel: string, message: string): Promise<[string, string]>;
+    ban(channel: string, username: string, reason: string): Promise<[string, string, string]>;
+    clear(channel: string): Promise<[string]>;
+    color(channel: string, newColor: string): Promise<[string]>;
```

Comment:
This seems incorrect too https://github.com/twitch-apis/twitch-js/blob/master/docs/Chat/Commands.md#color

Reply:
In this case the documentation is actually wrong. I'll submit a PR there for it. From the source: `function color(channel, newColor)`